### PR TITLE
fix(button): Update button to use semibold weight

### PR DESCRIPTION
**Pull request by notandrew**
Update button text to be semibold instead of regular font weight. Fix https://github.com/stardust-ui/react/issues/862

**Before**
image

**After**
image

### CODE REVIEW

File: src/themes/teams/components/Button/buttonStyles.ts [hunk 1]
```diff
@@ -279,10 +279,12 @@ const buttonStyles: ComponentSlotStylesInput<ButtonProps & ButtonState, any> = {
     }
   },
 
-  content: () => ({
+  // modifies the text of the button
+  content: ({ variables }): ICSSInJSStyle => ({
     overflow: 'hidden',
     textOverflow: 'ellipsis',
     whiteSpace: 'nowrap',
+    fontWeight: variables.fontWeight,
```
Comment (codepretty) [inline src/themes/teams/components/Button/buttonStyles.ts]: i think you had mentioned this before, but suggest renaming `variables.fontWeight` to something more specific

Comment (kuzhelov) [reply src/themes/teams/components/Button/buttonStyles.ts]: `buttonFontWeight` for now - would it be better?

File: src/themes/teams/components/Button/buttonVariables.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export interface ButtonVariables {
   borderRadius: string
   circularRadius: string
   paddingLeftRightValue: number
+  fontWeight: string
```
Comment (kuzhelov) [inline src/themes/teams/components/Button/buttonVariables.ts]: shouldn't the value be provided for this variable (or am I missing something?)

Comment (codepretty) [reply src/themes/teams/components/Button/buttonVariables.ts]: The value was already provided - https://github.com/stardust-ui/react/blob/master/packages/react/src/themes/teams/components/Button/buttonVariables.ts

File: packages/react/src/themes/teams/components/Button/buttonVariables.ts [hunk 3]
```diff
@@ -74,7 +75,7 @@ export default (siteVars: any): ButtonVariables => {
     maxWidth: pxToRem(280),
     borderRadius: pxToRem(2),
     circularRadius: pxToRem(999),
-    fontWeight: siteVars.fontWeightSemibold,
+    buttonFontWeight: siteVars.fontWeightSemibold,
```
Comment (mnajdova) [inline packages/react/src/themes/teams/components/Button/buttonVariables.ts]: I know there was already discussion about the name, but maybe `contentFontWeight` will be better name, as it is not applied to the icon part. In my opinion` buttonFontWeight` is not any different then `fontWeight`, as it is used inside the button, but doesn't explain that it is applied only in the content.

**Review by mnajdova — APPROVED** [review]

**Pull request merged**